src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -1,24 +1,57 @@
-import {NgModule, OnInit, Directive, ElementRef, HostBinding} from '@angular/core';
+import {NgModule, OnInit, Directive, ElementRef, HostBinding, OnDestroy} from '@angular/core';
+import {Event, Router, NavigationEnd, ResolveEnd} from '@angular/router';
+import {filter, map} from 'rxjs/operators';
+import {Subscription} from "rxjs";
 
 /** The timeout id of the previous focus change. */
 let lastTimeoutId = -1;
 
 @Directive({
   selector: '[focusOnNavigation]',
 })
-export class NavigationFocus implements OnInit {
+export class NavigationFocus implements OnInit, OnDestroy {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  private softNavFocus = false;
+  private subscriptions = new Subscription();
+
+  constructor(private el: ElementRef, private router: Router) {
+    // subscribing to events in constructor to catch the correct NavigationEnd/ResolveEnd from previous page
+    // and determine whether the element in question needs to be focused
+    // switching between different components/cdk produces a ResolveEnd instead of NavigationEnd event
+    // the focus is set in ngOnInit to account for late loaded content
+    this.subscriptions.add(
+      this.router.events
+        .pipe(
+          filter((event: Event) => event instanceof NavigationEnd || event instanceof ResolveEnd),
+          map(navEnd => navEnd as NavigationEnd))
+      .subscribe((navEnd: NavigationEnd) => {
+        const currentLocation = new URL(window.location.href);
+        if (!currentLocation.hash && isSoftNav(navEnd)) {
+          this.softNavFocus = true;
+          // navigating between CDK/Components pages only runs ngOnInit once so the focus must also be set here
+          this.el.nativeElement.focus({preventScroll: true});
+        }
+      }))
+  }
 
   ngOnInit() {
-    clearTimeout(lastTimeoutId);
-    // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus({preventScroll: true}),
-      100);
+    if (this.softNavFocus) {
+      // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
+      clearTimeout(lastTimeoutId);
+      lastTimeoutId = window.setTimeout(()=>this.el.nativeElement.focus({preventScroll: true}), 100);
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.unsubscribe();
   }
 }
 
+function isSoftNav(navigationEnd: NavigationEnd) {
+  return navigationEnd.id !== 1;
```

Comment:
I thought @atscott said `0` was a hardnav? How did you get `1`, is that just from experimenting?

Reply:
I confirmed with him and he said 0 is actually filtered out so it should be 1
https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L448

Reply:
Yea, sorry about that. The `initialNavigation` would be `1`, as the `id` is incremented from `0` to `1` before calling `transitions.next`: https://github.com/angular/angular/blob/master/packages/router/src/router.ts#L1151. To be honest, I don't even know if that filter is necessary in the router pipe (it may have been at some point in the past).

Reply:
Ah ok, do you think this is something worth having a more official API for? If so maybe @vxlydia can send a PR for it

Reply:
That could be useful. Maybe something simple like: "each navigation has a unique id and is available on the RouterEvents" and potentially something about how the `id` is a number that's incremented. I'd be wary of making any promises about the meaning of the id beyond that.

Reply:
Yeah that was kind of my point, making an assumption about the id _is_ hacky, you already pointed out that the filter might be unnecessary and if we remove it we'd need to change this check to compare to `0` instead of `1`. I was thinking more like adding a property to the event that can be more reliably checked against in this type of situation (I'm sure there's many ways we could accomplish this, but for example we could add `{from: Route|null, to: Router|null}` - that way a simple `if(event.from)` would tell us if its a softnav)

Reply:
Oh, sorry - I read that as "official docs" rather than "official API". Yes, there are a lot of holes in the router APIs but I can't necessarily recommend spending too much time on new features since the plan for the future of the router is unclear. Adding more API surface that we have to maintain through a potential refactor may be undesirable.

That said, I would absolutely file an issue for this and I can triage it as a feature request. Then we can also see if it gains community support.